### PR TITLE
SMHE-1335: Jobs should also be scheduled with a delay

### PR DESCRIPTION
**Pull request by kroesctrl**

### CODE REVIEW

File: osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java [hunk 1]
```diff
@@ -127,6 +130,30 @@ public Trigger createJobTrigger(
         .build();
   }
 
+  /**
+   * Create a {@link Trigger} instance using a {@link JobDetail} instance and a delay in seconds.
```
Comment (smvdheijden) [inline osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java]: Delay could probably be any interval unit, not just seconds, as the interval unit is also passed as a parameter?
```suggestion
   * Create a {@link Trigger} instance using a {@link JobDetail} instance and a delay.
```

File: osgp/shared/shared/src/test/java/org/opensmartgridplatform/shared/application/config/scheduling/OsgpSchedulerTest.java [hunk 2]
```diff
@@ -118,4 +133,27 @@ CronTrigger assertQuartzTrigger(
...
+        .isEqualTo(jobClazz.getSimpleName());
+    verify(this.quartzScheduler)
+        .scheduleJob(
+            this.jobDetail2ArgumentCaptor.capture(),
+            this.jobTriggerSetArgumentCaptor.capture(),
+            eq(true));
+    assertThat(this.jobDetail2ArgumentCaptor.getValue())
+        .isEqualTo(this.jobDetailArgumentCaptor.getValue());
+    assertThat(this.jobTriggerSetArgumentCaptor.getValue().size()).isOne();
+    assertThat(this.jobTriggerSetArgumentCaptor.getValue().size()).isOne();
```
Comment (smvdheijden) [inline osgp/shared/shared/src/test/java/org/opensmartgridplatform/shared/application/config/scheduling/OsgpSchedulerTest.java]: Twice the same assertion.

File: osgp/shared/shared/src/test/java/org/opensmartgridplatform/shared/application/config/scheduling/OsgpSchedulerTest.java [hunk 3]
```diff
@@ -118,4 +133,27 @@ CronTrigger assertQuartzTrigger(
...
+    final SimpleTrigger simpleTrigger =
+        (SimpleTrigger) this.jobTriggerSetArgumentCaptor.getValue().iterator().next();
+    assertThat(simpleTrigger.getStartTime())
+        .isCloseTo(new DateTime().plusSeconds(delayInSeconds).toDate(), 100l);
+
+    return simpleTrigger;
```
Comment (smvdheijden) [inline osgp/shared/shared/src/test/java/org/opensmartgridplatform/shared/application/config/scheduling/OsgpSchedulerTest.java]: Seems to me this method could use some refactoring to increase readability? 

Furthermore assertions on `simpleTrigger` are performed in this method, but also in the calling `testCreateAndScheduleSimpleJobDataMap` method?

**Review by smvdheijden — CHANGES_REQUESTED** [review]

File: osgp/shared/shared/src/test/java/org/opensmartgridplatform/shared/application/config/scheduling/OsgpSchedulerTest.java (hunk 2 above)
Comment (kroesctrl) [reply osgp/shared/shared/src/test/java/org/opensmartgridplatform/shared/application/config/scheduling/OsgpSchedulerTest.java]: removed it

File: osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java (hunk 1 above)
Comment (kroesctrl) [reply osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java]: Changed it

File: osgp/shared/shared/src/test/java/org/opensmartgridplatform/shared/application/config/scheduling/OsgpSchedulerTest.java (hunk 3 above)
Comment (kroesctrl) [reply osgp/shared/shared/src/test/java/org/opensmartgridplatform/shared/application/config/scheduling/OsgpSchedulerTest.java]: I thought i could stand on the shoulder of giants, but i was wrong. ;-)
Refactored the testing methods

**Review by smvdheijden — APPROVED** [review]

**Review by natasja-n — APPROVED** [review]

**Pull request merged**